contracts/JBSingleTokenPaymentTerminalStore3_2.sol

```diff
@@ -511,25 +516,23 @@ contract JBSingleTokenPaymentTerminalStore3_2 is
     // The new total amount that has been distributed during this funding cycle.
     uint256 _newUsedDistributionLimitOf = usedDistributionLimitOf[
       IJBSingleTokenPaymentTerminal(msg.sender)
-    ][_projectId][fundingCycle.number] + _amount;
+    ][_projectId][fundingCycle.number][_currency] + _amount;
 
     // Amount must be within what is still distributable.
-    (uint256 _distributionLimitOf, uint256 _distributionLimitCurrencyOf) = IJBController3_2(
-      directory.controllerOf(_projectId)
-    ).fundAccessConstraintsStore().distributionLimitOf(
+    uint256 _distributionLimit = IJBController3_2(directory.controllerOf(_projectId))
+      .fundAccessConstraintsStore()
+      .distributionLimitOf(
         _projectId,
         fundingCycle.configuration,
         IJBSingleTokenPaymentTerminal(msg.sender),
-        IJBSingleTokenPaymentTerminal(msg.sender).token()
+        IJBSingleTokenPaymentTerminal(msg.sender).token(),
+        _currency
       );
 
     // Make sure the new used amount is within the distribution limit.
-    if (_newUsedDistributionLimitOf > _distributionLimitOf || _distributionLimitOf == 0)
+    if (_newUsedDistributionLimitOf > _distributionLimit || _distributionLimit == 0)
       revert DISTRIBUTION_AMOUNT_LIMIT_REACHED();
 
-    // Make sure the currencies match.
-    if (_currency != _distributionLimitCurrencyOf) revert CURRENCY_MISMATCH();
```

Comment:
Note for reviewers: no need to check this any longer since the distributionLimit for _currency is explicitly sought after.